pkg/lint/rules/values.go

```diff
@@ -28,7 +28,7 @@ import (
 )
 
 // Values lints a chart's values.yaml file.
-func Values(linter *support.Linter) {
+func Values(linter *support.Linter, values map[string]interface{}) {
```

Comment:
As per our semver contract, we cannot change function signatures. Users consuming this function in their codebases would break when they upgrade from 3.0 to 3.1. General rule is "code that used to work before will continue to work in future releases".

Can you try to find another way to resolve this without breaking backwards compatibility?

Reply:
That makes sense. Is changing the internals of exposed functions okay as I've done [here](https://github.com/helm/helm/pull/7417/files#diff-432bdc8e6c573117062cb071cdd1aaa6R33) okay? If so, I think exposing a new function that I call there while leaving the existing function intact should work. Also, what do you think about [my thoughts below](https://github.com/helm/helm/pull/7417/files#r367946663)

Reply:
As long as the expected behaviour is the same (that is, if you provide the same input, you'll receive the same output), then yes.

Reply:
Hmmm, there's definitely room for interpretation around the work 'expected'. Given that this is a bug, the current behavior isn't really the expected behavior. Today, if someone were to call `pkg/lint/All(...)` with values that satisfy the requirements of JSON schema that are not satisfied by the chart's builtin values, the linter would return errors. That being said, _this is unexpected behavior_ as described in the pull request and related issue. With the change I've made to the internals of the function, the linter would no longer return any errors - which is a different output with the same input, but yields the desired behavior